graph/src/components/store/traits.rs

```diff
@@ -156,17 +156,17 @@ pub trait SubgraphStore: Send + Sync + 'static {
 #[async_trait]
 pub trait WritableStore: Send + Sync + 'static {
     /// Get a pointer to the most recently processed block in the subgraph.
-    fn block_ptr(&self) -> Option<BlockPtr>;
+    async fn block_ptr(&self) -> Option<BlockPtr>;
 
     /// Returns the Firehose `cursor` this deployment is currently at in the block stream of events. This
     /// is used when re-connecting a Firehose stream to start back exactly where we left off.
-    fn block_cursor(&self) -> Option<String>;
+    async fn block_cursor(&self) -> Option<String>;
 
     /// Deletes the current Firehose `cursor` this deployment is currently at.
-    fn delete_block_cursor(&self) -> Result<(), StoreError>;
+    async fn delete_block_cursor(&self) -> Result<(), StoreError>;
 
     /// Start an existing subgraph deployment.
-    fn start_subgraph_deployment(&self, logger: &Logger) -> Result<(), StoreError>;
+    async fn start_subgraph_deployment(&self, logger: &Logger) -> Result<(), StoreError>;
```

Comment:
I generally prefer to handle this type of concurrency on the caller, this is not really about the async

Reply:
Please expand because I do not get what you mean here.

Reply:
What I'm suggesting is that we don't make this particular one async, as in, let this call be blocking and let the caller decide whether do send it to a task or block or it. @leoyvens any preference about this one?

Reply:
On the general pattern we want to follow, the most important thing is to avoid doing blocking operations within a non-blocking task, which is an easy mistake to make when the store exposes blocking fns. So it's safest to expose async functions when in doubt.

But in this case I agree it's deep rabbit hole for a function that has few call sites, so this is fine.

Reply:
I achieved a good chunk and I reached the point where I need to do something so that  `self.store.primary_conn()` would not block the whole thread. I wasn't sure so decided to stop there. Would you have any insights, think some ground work would be need for the `primay_conn` stuff, is it possible?

I'm willing to maybe open a new PR early next week to convert even more stuff up.